pallets/subtensor/src/subnets/weights.rs

```diff
@@ -452,50 +463,29 @@ impl<T: Config> Pallet<T> {
         uids.len() <= subnetwork_n as usize
     }
 
-    #[allow(clippy::arithmetic_side_effects)]
-    pub fn can_commit(netuid: u16, who: &T::AccountId) -> bool {
-        if let Some((_hash, commit_block)) = WeightCommits::<T>::get(netuid, who) {
-            let interval: u64 = Self::get_commit_reveal_weights_interval(netuid);
-            if interval == 0 {
-                return true; //prevent division by 0
-            }
-
+    ///
+    /// This function checks if the `weights_rate_limit` time has passed since the last
+    /// committed weight. If it has, the validator can commit new weights.
+    ///
+    pub fn check_rate_limit_for_commit(netuid: u16, who: &T::AccountId) -> bool {
```

Comment:
there should be one function which works for both set_weights and commit_weights. If there are two, then sooner or later they will start doing different things.

Reply:
It's related to the commit part, do_set_weight function is difference one so it doesn't make sense for me to add this to do_set_weight.

Reply:
where it's repeated? I guess it's already done as one function.